### ISSUE TITLE
Workspace default container setting being overridden/ignored by sync

### ISSUE THREAD

**Issue by sjclayton** [zen-browser/desktop#2154]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

Ok, here is the scenario... I have 2 machines I use regularly, one Linux (primary) and one Windows (work).

I have 4 workspaces set up (eg, Workspace A,B,C,D) on both, they are synced with the workspace sync option. I also have 4 containers set up (eg, Container A,B,C,D), they are identical on both machines, names/colors and icons.

I set Workspace C's default container to Container D on my Linux machine, it removes the default container set on my Windows laptop for Workspace C (also set to Container D), if I then manually set Container D as the default container for Workspace C on this Windows machine, then it clears it from being set on my Linux machine. 

So basically which ever machine I set it on clears the setting on the other. 

**Expected Result:**

Both machines have the same named workspace with the same default container set.

### Reproducible?

- [ ] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.10

### What platform are you seeing the problem on?

Linux, Windows

**Comment by niklaswimmer** [comment]
Hey, wild guess of mine, but do you sync (as in Firefox's sync feature) your containers or have you just created containers with the same name on both machines? If so, I would suspect (no clue how it really works -> a wild guess) that Zen stores the container ID, which differs between the machines and that it clears the container field if it can not one with the same ID locally.

**Comment by sjclayton** [comment]
@niklaswimmer 

This is supposed to be fixed by the pull request mentioned above, so the behaviour is clearly something that was not implemented in the first place, which is why it wasn't working.

**Comment by niklaswimmer** [comment]
Ups, I don't know how I missed that linked PR.

However, maybe I misunderstand something, but if by Containers we mean the Multi-Account Container extension, then syncing them is already possible and can be enabled in the extension's settings (although I gotta admit the *only* reason why *I* am in this thread is because that was not working particularly well... so I am excited to see another solution being developed).

**Comment by PseudoMortus** [comment]
Attempting top reopen the issue, as not only is it still relevant, but deleting the problematic container and workspace and remaking them in multiple configurations yielded no results either. Have syncing on the browser level AND multi container level, both on Linux, and has been a problem for be for a few months now at least.

**Comment by Propheticus** [comment]
What has worked pretty well for me so far:
- Creating / assigning containers on 1 device. So not making containers on both with the same name (while their name is the same, their ID would differ)
- Disable sync of "settings"

<img width="571" height="186" alt="Image" src="https://github.com/user-attachments/assets/d920e6e7-269e-468b-a0b8-a9e8183b1b9b" />

- Enable the somewhat hidden sync within Multi account container extension settings:

<img width="571" height="114" alt="Image" src="https://github.com/user-attachments/assets/e7266b2d-ddff-474e-a4c3-314b3e40fb97" />

**Comment by PseudoMortus** [comment]
> What has worked pretty well for me so far:
> 
>     * Creating / assigning containers on 1 device. So not making containers on both with the same name (while their name is the same, their ID would differ)
> 
>     * Disable sync of "settings"
> 
> 
> <img alt="Image" width="571" height="186" src="https://private-user-images.githubusercontent.com/6628064/514335553-d920e6e7-269e-468b-a0b8-a9e8183b1b9b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMxMjM4NTAsIm5iZiI6MTc2MzEyMzU1MCwicGF0aCI6Ii82NjI4MDY0LzUxNDMzNTU1My1kOTIwZTZlNy0yNjllLTQ2OGItYTBiOC1hOWU4MTgzYjFiOWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMTRUMTIzMjMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmQ1NTdjOGU4NzI0YTEzNjI3MjBkYzliODExNTdiODE5ZmZjZTlmZGZlN2ZlNGQ3M2Q3ZmQ4MTcxNWM2M2VlNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8m2ADv4t6eThFpK6rWdQ3W4h_WgE6t_N_PvQgkp2bk8">
> 
>     * Enable the somewhat hidden sync within Multi account container extension settings:
> 
> 
> <img alt="Image" width="571" height="114" src="https://private-user-images.githubusercontent.com/6628064/514335791-e7266b2d-ddff-474e-a4c3-314b3e40fb97.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMxMjM4NTAsIm5iZiI6MTc2MzEyMzU1MCwicGF0aCI6Ii82NjI4MDY0LzUxNDMzNTc5MS1lNzI2NmIyZC1kZGZmLTQ3NGUtYTRjMy0zMTRiM2U0MGZiOTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTExNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTExMTRUMTIzMjMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGUzYzBhNDZkMGM1ZmFhMzRhNzFiZmU0MzFlYThiNGVjMzU5NGMzMzI5YTkwMTdhYmIyNzcwOWIxN2UyMTBiNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.2ZqeNBCccQr1xiFBp7pfTgKSTcw9Tgtt-vLYM6eg7EI">

How many container workspaces do you tend to run? I currently have 6, and it's just this one container workspace that's borking, even after this suggestion, so I'm curious if quantity plays a part

**Comment by kkiris** [comment]
I managed to fix this issue on my devices through the following steps: 
1. Go to `about:profiles` on both devices, then click to open the profile's Root Directory.
2. Fully close Zen Browser on both devices. 
3. Copy `containers.json` from the primary device to the other device. 
4. Reopen Zen Browser on the primary device and verify correct container assignments. 
5. Reopen Zen Browser on the other device and verify correct container assignments. 
---
When containers are synced across devices, it copies the names, colors, and icons, but not the IDs. By copying `containers.json`, you ensure the container IDs match between devices.
